bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/util/AsyncBackOff.java

```diff
@@ -0,0 +1,83 @@
+package com.google.cloud.bigtable.util;
+
+import com.google.api.client.util.BackOff;
+import com.google.api.client.util.ExponentialBackOff;
+import com.google.cloud.bigtable.config.Logger;
+import com.google.cloud.bigtable.grpc.io.RefreshingOAuth2CredentialsInterceptor;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Stopwatch;
+import com.google.common.base.Ticker;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Wrapper around {@link ExponentialBackOff} to allow it to be used w/o a sleeper.
+ * Its marriage of a rate limiter and an ExponentialBackOff.
+ * This class is not threadsafe.
+ */
+public class AsyncBackOff {
+  private static final Logger LOG = new Logger(AsyncBackOff.class);
+
+  private final Stopwatch stopwatch;
```

Comment:
I can revert it back to rate limiting, but I think the ramp up from exponential backoff is better